### PR TITLE
Documentation updates and exporting types and prop interfaces.

### PR DESCRIPTION
**Pull request by EdwinGuzman**
No JIRA ticket

## This PR does the following:

- Various documentation updates and updating links and syntax.

### Bigger updates

- The pull request template now includes an `Accessibility Checklist` section and an `Open Questions` section.
- The README had the most changes.
  - Removes recommendations that are no longer relevant.
  - Update example code syntax.
  - Removes the "npm link" section for local development because I'm pretty sure that no longer works with node 18. We now recommend release candidates.
  - Removes the unpkg references to the older DS versions since we should really just support v3.
- Exports all types and props from components. Some apps need various ones and we might as well export all of them now.

### Minor updates

- For links, @bigfishdesign13 I updated the vercel links for the V2 and V1 instances to the github pages link. So from `https://nypl-design-system-git-reservoir-v216-nypl.vercel.app` to `https://nypl.github.io/nypl-design-system/reservoir/v2`. Let me know what you think.
- 

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -6,27 +6,42 @@ Fixes JIRA ticket xxx
...
+-
 
 ## Accessibility concerns or updates
 
-<!--- Describe any accessibility concerns or updates that were made that should be known. -->
+<!-- Describe any accessibility concerns or updates that were made that should be known. -->
+
+-
+
+### Accessibility Checklist
```
Comment (bigfishdesign13) [inline .github/PULL_REQUEST_TEMPLATE:19]: Add a checkbox about color contrast.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I have just one suggestion.  Everything else looks good.  That said, how did you ensure that all `interfaces` were exported properly?  Global search?

**Comment by EdwinGuzman** [comment]
> how did you ensure that all interfaces were exported properly? 

Went through all the files... but I might've missed some as per human error.

- Added a checkbox for color contrast but pointed to Storybook's accessibility tab to point to where and how to test for it.

File: .github/PULL_REQUEST_TEMPLATE (hunk 1 above)
Comment (bigfishdesign13) [reply .github/PULL_REQUEST_TEMPLATE:19]: I like what you added.  That's what I assumed would be the approach for checking the color contrast.

Comment (EdwinGuzman) [reply .github/PULL_REQUEST_TEMPLATE:19]: I'd also prefer to keep devs in Storybook for a11y checking.

**Review by bigfishdesign13 — APPROVED** [review]
⚡

**Review by jackiequach — COMMENTED** [review]
Missing ButtonProps, and new component types/props: FilterBarInlineProps, FilterBarItemsType, NewsletterSignupProps

**Comment by EdwinGuzman** [comment]
@jackiequach good catch, updated

**Review by jackiequach — APPROVED** [review]

**Review by 7emansell — APPROVED** [review]

**Pull request merged**